File: Difficulty: Basic/Bitwise Right Shift/bitwise-right-shift.java
//{ Driver Code Starts

// } Driver Code Ends

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        // Create a Scanner object to read input from the console
        Scanner scanner = new Scanner(System.in);

        // Read two integer values from the user
        int number = scanner.nextInt();
        int shiftAmount = scanner.nextInt();

        // Perform right shift operation and print the result
        System.out.println(number >> shiftAmount);

        // Close the scanner to prevent resource leaks
        scanner.close();
    }
}


//{ Driver Code Starts.
// } Driver Code Ends

File: Difficulty: Basic/Bitwise Right Shift/bitwise-right-shift.cs
//{ Driver Code Starts
// Initial Template for C#

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Numerics;

namespace DriverCode {

class GFG {
    static void Main(string[] args) {
        int testcases; // Taking testcase as input
        testcases = Convert.ToInt32(Console.ReadLine());
        while (testcases-- > 0) // Looping through all testcases
        {
            var ip = Console.ReadLine().Trim().Split(' ');
            int a = int.Parse(ip[0]);
            int b = int.Parse(ip[1]);
            Solution obj = new Solution();
            obj.rightShift(a, b);

            Console.Write("~" + "\n");
        }
    }
}

}
// } Driver Code Ends

// User function Template for C#
class Solution {
    // Complete this function
    public void rightShift(int a, int b) {
        // Perform right shift operation
        int ans = a >> b;

        // print the result
        Console.WriteLine(ans);
    }
}
